app/mailers/invoice_mailer.rb

```diff
@@ -34,8 +35,10 @@ def set_text(organization, sale, mail_template)
     if !mail_template.text.include? "#"
       return mail_template.text
     elsif sale.user.contacts.present?
-      contact = organization.contacts.find(sale.user.contacts)
-      return set_variables(mail_template.text, Contact::ALLOWED_COLUMNS, contact.attributes)
+      sale_fields = sale.attributes
+      contact_fields = organization.contacts.find(sale.user.contacts).attributes
+      fields = sale_fields.merge(contact_fields)
```

Comment:
Hm... wouldn't this override the variables from contact_field?

I propose to use something like `ALLOWED_COLUMNS = ['#recipient_name', '#sender_name', '#sender_telephone' ...]` to make it extra clear for the user that will type in the variable *names* in their mail template.